CHANGES.md

```diff
@@ -1,6 +1,473 @@
+* added support for reduce intents on coforall+on loops
+* added support for array return types with unspecified domains / element types
+  (e.g., `proc foo(): []` says that foo() returns an array but little else)
+* added support for dynamic dispatch on variadic (varargs) methods
+* added support for rectangular domains and arrays of `enum` or `bool` idxType
+* updated `--memStats` to work properly in multi-locale runs
+  (see https://chapel-lang.org/docs/master/modules/standard/Memory.html#Memory.printMemAllocStats)
+* added subtyping for owned and shared classes
+  (see https://chapel-lang.org/docs/master/builtins/internal/SharedObject.html#coercions-for-shared and
+   https://chapel-lang.org/docs/master/builtins/internal/OwnedObject.html#coercions-for-owned))
+* added support for assigning `nil` to owned and shared classes
+* made the default intent for formals declared with owned/shared type `const in`
+* made forall loops borrow outer owned and shared class objects by default
+* improved the appearance of shared objects printed via writeln()
+* added support for argument types containing nested generic or query types
+  (see 'Formal Arguments of Partially Generic Type' in the 'Generics' chapter)
+* improved the instantiation of generic arguments when coercions are involved
+* added support for infix bool operators to throw
+* improved isSubtype() and isProperSubtype() to handle cases that can coerce
+* extended isSubtype() and isProperSubtype() to work with generic types
+  (see https://chapel-lang.org/docs/master/builtins/internal/UtilMisc_forDocs.html#UtilMisc_forDocs.isSubtype)
+* improved disambiguation and visibility rules when resolving function calls
+  (see 'Function Visibility in Generic Functions' in the 'Generics' chapter)
+* made `in` arguments more consistent with variable initialization
+  (see 'Argument Intents' in the language specification's 'Procedures' chapter)
+  (see https://github.com/chapel-lang/chapel/blob/master/doc/rst/developer/chips/13.rst)
+* added support for casts from `c_void_ptr` to `c_ptr`
+
+Removed Features
+----------------
+* removed implicit `outer` field on nested types with initializers
+* removed the `UtilReplicatedVar` module (deprecated in 1.17)
+* disabled non-`ref` forall intents when using recursive parallel iterators
+
+Standard Modules / Library
+--------------------------
+* increased the use of error-handling in standard modules
+* deprecated 'out error' patterns in libraries in favor of throwing versions
+* converted some non-fatal checks from using `halt()` to `warning()`
+* deprecated `Owned(C)` and `Shared(C)` in favor of `owned C` and `shared C`
+  (see 'Class Types' in the 'Classes' chapter of the language specification))
+* added type query routines for `owned`, `shared`, `unmanaged`, and `borrowed`
+  (see https://chapel-lang.org/docs/master/modules/standard/Types.html#Types.isOwnedClass)
+* added support for singular value decomposition to the `LinearAlgebra` module
```

Comment:
The following block should actually be in "Package Modules":

```
* added support for singular value decomposition to the `LinearAlgebra` module
  (see https://chapel-lang.org/docs/master/modules/packages/LinearAlgebra.html#LinearAlgebra.svd)
* enabled using `LinearAlgebra` without having `BLAS/LAPACK` installed
  (see https://chapel-lang.org/docs/master/modules/packages/LinearAlgebra.html#compiling-with-linear-algebra)
* added configs `blasImpl` and `blasHeader` to `BLAS` to select backend
  (see https://chapel-lang.org/docs/master/modules/packages/BLAS.html#BLAS.blasImpl)
* added configs `lapackImpl` and `lapackHeader` to `LAPACK` to select backend
  (see https://chapel-lang.org/docs/master/modules/packages/LAPACK.html#LAPACK.lapackImpl)
* deprecated `BLAS` config `isBLAS_MKL` in favor of using `blasImpl`
* made LinearAlgebra sparse matrices store indices in unsorted order
```

That was probably my mistake - sorry!